src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Utilities/ExportFactory/ProjectFileModelWatcherFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.VS.Editor;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Utilities.ExportFactory
+{
+    [Export(typeof(IExportFactory<IProjectFileModelWatcher>))]
+    internal class ProjectFileModelWatcherFactory : IExportFactory<IProjectFileModelWatcher>
```

Comment:
Actually, I think I can make a generic implementation by having a series of static methods that return an IExportFactory of the correct type for each type that we need:

```C#
    internal class GenericExportFactory<T> : IExportFactory<T>
    {
        [Export(typeof(IExportFactory<EditProjectFileVsFrameEvents>))]
        public static IExportFactory<EditProjectFileVsFrameEvents> EditProjectFileVsFrameEventsExportFactory(
            ExportFactory<EditProjectFileVsFrameEvents> factory)
        {
            return new GenericExportFactory<EditProjectFileVsFrameEvents>(factory);
        }

        private readonly ExportFactory<T> _factory;

        private GenericExportFactory(ExportFactory<T> factory)
        {
            _factory = factory;
        }

        public T CreateExport()
        {
            throw new NotImplementedException();
        }
    }
```
Which would be preferable? Or do you have another solution for mocking ExportFactory?